### PR TITLE
[DOC] #4674 - Add more documentation for @api decorator.

### PR DESCRIPTION
**Pull request by fantasya-pbem**
Have I explained the stuff correctly?

### CODE REVIEW

File: src/borg/remote.py [hunk 1]
```diff
@@ -423,9 +423,14 @@ def api(*, since, **kwargs_decorator):
     """Check version requirements and use self.call to do the remote method call.
 
     <since> specifies the version in which borg introduced this method,
-    calling this method when connected to an older version will fail without transmiting
+    calling this method when connected to an older version will fail without transmitting
     anything to the server.
 
+    <previously> is the value resulting in the behaviour before introducing the new version.
```
Comment (ThomasWaldmann) [inline src/borg/remote.py]: ... the new parameter.

File: src/borg/remote.py [hunk 2]
```diff
@@ -423,9 +423,14 @@ def api(*, since, **kwargs_decorator):
...
+
+    <dontcare> is a flag to set the behaviour if an old version is called the new way. If set to True, the method is
+    called as defined by <previously>. If False, an exception is generated.
+
```
Comment (ThomasWaldmann) [inline src/borg/remote.py]: ... the method is called without the (not yet supported) parameter (this should be done if that is the more desirable behaviour).

File: src/borg/remote.py [hunk 3]
```diff
@@ -422,16 +422,20 @@ def write(self, fd, to_send):
...
-    If a previous hardcoded behaviour is parameterized in a version, this allows calls that
-    use the previously hardcoded behaviour to pass through and generates an error if another
-    behaviour is requested by the client.
+    If a previous hardcoded behaviour is parameterized in a version, this allows calls that use the previously
+    hardcoded behaviour to pass through and generates an error if another behaviour is requested by the client.
 
-    e.g. when 'append_only' was introduced in 1.0.7 the previous behaviour was what now is append_only=False.
+    E.g. when 'append_only' was introduced in 1.0.7 the previous behaviour was what now is append_only=False.
     Thus @api(..., append_only={'since': parse_version('1.0.7'), 'previously': False}) allows calls
     with append_only=False for all version but rejects calls using append_only=True on versions older than 1.0.7.
```
Comment (ThomasWaldmann) [inline src/borg/remote.py]: guess it rather should be somewhere here?

**Comment by fantasya-pbem** [comment]
Shall we close this PR?

**Comment by ThomasWaldmann** [comment]
No, rather fix it, see my comments.

**Comment by ThomasWaldmann** [comment]
Thanks!

**Pull request merged**